@here/harp-mapview/lib/Utils.ts

```diff
@@ -446,6 +460,229 @@ export namespace MapViewUtils {
         return bytes;
     }
 
+    /**
+     * Computes estimate for size of a THREE.Object3D object and its children. Shared materials
+     * and/or attributes will be counted multiple times.
+     *
+     * @param object The mesh object to evaluate
+     * @param size The [[MemoryUsage]] to update.
+     *
+     * @returns Estimate of object size in bytes for heap and GPU.
+     */
+    export function estimateObject3dSize(
+        object: THREE.Object3D,
+        parentSize?: MemoryUsage
+    ): MemoryUsage {
+        const size =
+            parentSize !== undefined
+                ? parentSize
+                : {
+                      heapSize: 0,
+                      gpuSize: 0
+                  };
+
+        estimateMeshSize(object, size);
+
+        if (object.children.length > 0) {
+            for (const child of object.children) {
+                estimateObject3dSize(child, size);
+            }
+        }
+        return size;
+    }
+
+    function estimateTextureSize(
+        texture: THREE.Texture | null,
+        isVisible: boolean,
+        objectSize: MemoryUsage
+    ) {
+        if (texture === null || texture.image === undefined) {
+            return;
+        }
+
+        // May be HTMLImage or ImageData
+        const image = texture.image;
+        // Assuming RGBA
+        objectSize.heapSize += 4 * image.width * image.height;
+        if (isVisible) {
+            objectSize.gpuSize += 4 * image.width * image.height;
+        }
+    }
+
+    function estimateMaterialSize(
+        material: THREE.Material,
+        isVisible: boolean,
+        objectSize: MemoryUsage
+    ) {
+        if (
+            material instanceof THREE.RawShaderMaterial ||
+            material instanceof THREE.ShaderMaterial
+        ) {
+            const rawMaterial = material;
+            for (const name in rawMaterial.uniforms) {
+                if (rawMaterial.uniforms[name] !== undefined) {
+                    const uniform = rawMaterial.uniforms[name];
+                    if (uniform instanceof THREE.Texture) {
+                        estimateTextureSize(uniform, isVisible, objectSize);
+                    }
+                }
+            }
+        } else if (
+            material instanceof THREE.MeshBasicMaterial ||
+            material instanceof MapMeshBasicMaterial
+        ) {
+            const meshMaterial = material;
+            estimateTextureSize(meshMaterial.map, isVisible, objectSize);
```

Comment:
Will it count one texture attached to several objects in several meshes as once ?

I don't see any tracking of already counted objects.

Just highlighting issue, not sure if it's iportant at all at this stage.

(we don't have shared textured anywhere, only in ImageCache)

Reply:
I will add code to count attributes/arraybuffers and textures only once.